server/src/api/contacts/contacts.service.ts

```diff
@@ -62,14 +62,14 @@ export const create = async (r: any): Promise<response> => {
       VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10)
       RETURNING *
       `,
-    values: [r.firstname, r.lastname, r.email, r.custaddress, r.phone, r.donorbadge, r.seatingaccom, r.newsletter, r.vip, r.volunteer_list],
+    values: [r.firstname, r.lastname, r.email, r.address, r.phone, r.donorbadge, r.seatingaccom, r.newsletter, r.vip, r.volunteer_list],
   };
   return await buildResponse(myQuery, 'POST');
 };
 
 export const remove = async (id: string): Promise<response> => {
   const myQuery = {
-    text: 'DELETE FROM contacts WHERE customerid = $1;',
+    text: 'DELETE FROM contacts WHERE contactid = $1;',
```

Comment:
This delete won't work as is because of the fk policy linking the `contacts` and `orders` tables. We don't want to delete order information but there should be an option to delete contact information in case a user doesn't want there data to be stored anymore. There is also functionality in the front-end for admins that calls the related delete endpoint. So to avoid a refactor and to give users at least some kind of option to have there data removed (even if they have to contact PPH to do so), my suggestion is to set all fields to `NULL` except the `custid`. Also, so that contacts that have been nulled out do not show up in reporting, we should add logic to only (for now) to the `findAll`  service function that only returns results that have a non-null email address since contacts are required to have an email address. This could probably be implemented directly into the query somehow, but it is more complex so I'm not sure exactly.

Reply:
I assume the front end handles the requirement for an email address at the moment?

Reply:
Yes, an email address is required when creating a contact along with some other fields